plugin/plugin.go

```diff
@@ -152,19 +147,113 @@ func (e *Ratelimit) Evaluate(r rules.RuleMetadata, tx rules.TransactionState) {
 	// prettyPrint(tx.Collection(variables.Rule).FindAll())
 	// prettyPrint(tx.Collection(variables.ResponseHeaders).FindAll())
 
-	// tx.Variables().All(func(v variables.RuleVariable, col collection.Collection) bool {
-	// 	prettyPrint(map[string]interface{}{"variable": col.Name(), "col": col.FindAll()})
-	// 	return true
-	// })
 }
 
 func (e *Ratelimit) Type() rules.ActionType {
 	return rules.ActionTypeNondisruptive
 }
 
-func prettyPrint(i interface{}) {
-	s, _ := json.MarshalIndent(i, "", "\t")
-	fmt.Println(string(s))
+func (e *Ratelimit) parseConfig(config string) error {
+	// acceptable keys
+	var err error
+
+	tokens := strings.Split(config, "&")
+
+	requiredValues := map[string]bool{
+		"zone[]": false,
+		"events": false,
+		"window": false,
+	}
+
+	for _, token := range tokens {
+		key, value, found := strings.Cut(token, "=")
+		if !found || key == "" || value == "" || strings.Count(value, "=") != 0 {
```

Comment:
`strings.Count(value, "=")` will count all occurrences, while you want to know if there is one or not. `string.Contains` might do the trick.